bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -557,10 +557,15 @@ private long getLedgerToRereplicateFromHierarchy(String parent, long depth)
             return -1;
         }
 
-        Collections.shuffle(children);
+        if (depth == 3) {
+            // Avoid workers want to get the same ledger lock.
+            Collections.shuffle(children);
+        } else {
+            Collections.sort(children);
```

Comment:
It will sort according to the `String` type order instead of the LedgerId long type order, which can't more sure the sorted list is ordered by ledger creation time.

Reply:
The node of `subTreeCache.getChildren(parent)`  is 4 hex bytes，` Collections.sort()` could get the correct order.